src/fiatExchanges/utils.tsx

```diff
@@ -291,3 +294,78 @@ export const filterProvidersByPaymentMethod = (
 ) => {
   return externalProviders?.find((quote) => quote.paymentMethods.includes(paymentMethod))
 }
+
+export interface ProviderSelectionAnalyticsData {
+  centralizedExchangesAvailable: boolean
+  totalOptions: number
+  paymentMethodsAvailable: Record<PaymentMethod, boolean>
+  transferCryptoAmount: string
+  lowestFeeKycRequired: boolean | undefined
+  lowestFeeCryptoAmount: string | undefined
+  lowestFeeProvider: string | undefined
+  lowestFeePaymentMethod: PaymentMethod | undefined
+}
+
+/**
+ * Get analytics data for provider selection.
+ *
+ * Used for cico_providers_quote_selected and cico_providers_exchanges_selected analytics events.
+ *
+ * @param normalizedQuotes
+ * @param exchangeRates
+ * @param legacyMobileMoneyProviders
+ * @param centralizedExchanges
+ * @param transferCryptoAmount
+ */
+export function getProviderSelectionAnalyticsData({
+  normalizedQuotes,
+  exchangeRates,
+  legacyMobileMoneyProviders,
+  centralizedExchanges,
+  transferCryptoAmount,
+}: {
+  normalizedQuotes: NormalizedQuote[]
+  exchangeRates: { [token in Currency]: string | null }
+  legacyMobileMoneyProviders?: LegacyMobileMoneyProvider[]
+  centralizedExchanges?: ExternalExchangeProvider[]
+  transferCryptoAmount: string
+}): ProviderSelectionAnalyticsData {
+  let lowestFeePaymentMethod: PaymentMethod | undefined = undefined
+  let lowestFeeProvider: string | undefined = undefined
+  let lowestFeeCryptoAmount: BigNumber | null = null
+  let lowestFeeKycRequired: boolean | undefined = undefined
+  const centralizedExchangesAvailable = !!centralizedExchanges && centralizedExchanges?.length > 0
+  const paymentMethodsAvailable: Record<PaymentMethod, boolean> = {
+    [PaymentMethod.Bank]: false,
+    [PaymentMethod.Card]: false,
+    [PaymentMethod.Coinbase]: false,
+    [PaymentMethod.MobileMoney]:
+      !!legacyMobileMoneyProviders && legacyMobileMoneyProviders.length > 0,
+  }
+
+  for (const quote of normalizedQuotes) {
+    paymentMethodsAvailable[quote.getPaymentMethod()] = true
+    const fee = quote.getFeeInCrypto(exchangeRates)
+    if (fee && (lowestFeeCryptoAmount === null || fee.isLessThan(lowestFeeCryptoAmount))) {
+      lowestFeeCryptoAmount = fee
+      lowestFeePaymentMethod = quote.getPaymentMethod()
+      lowestFeeProvider = quote.getProviderId()
+      lowestFeeKycRequired = !!quote.getKycInfo()
+    }
+  }
+
+  return {
+    transferCryptoAmount,
+    paymentMethodsAvailable,
+    lowestFeePaymentMethod,
+    lowestFeeProvider,
+    lowestFeeKycRequired,
+    centralizedExchangesAvailable,
+    lowestFeeCryptoAmount: lowestFeeCryptoAmount?.toFixed(2),
+    // counts centralized exchanges as single option, since that's how they appear on the Select Providers screen
+    totalOptions:
```

Comment:
should coinbase be counted here too?

Reply:
it is-- normalizedQuotes.length will cover it